product/specs/2022-01-views/04-ui-requirements-for-views.md

```diff
@@ -0,0 +1,78 @@
+---
+title: Product Specs
+description: Views: UI Requirements for Views
+published: true
+date: 2022-01-13T19:49:54Z
+tags: 
+editor: markdown
+dateCreated: 2022-01-13T19:49:54Z
+---
+
+Based on the models defined for [views](/product/specs/2022-01-views/02-modeling-views.md) and their [columns](/product/specs/2022-01-views/03-modeling-view-columns.md), the first version of Views should support the following features.
+
+I've marked the features we should support as "for alpha release", but I've also listed a few potential future features so that we can think about them while designing code architecture and UI/UX.
+
+## Query
+- **For alpha release**: Users should be able to see the query associated with a View after it has been loaded.
+- **Potential future features**:
+	- Users can edit the query associated with a View.
+	- Users can create a new View using a query.
+
+## Columns
+- **For alpha release**: Users should be able to see all columns associated with a view. Each column should show:
+	- Data Type (non-editable)
+	- Data Sources (editable if there's no Formula)
```

Comment:
I don't understand the goal of making the data sources editable.  Would the intention be to drop a different column in in place of the one shown?  Wouldn't it be simpler / smoother to remove that column and add a different one?

Reply:
I think that if users want to reuse a view w a different table, this might come in handy. I think removing/adding a new column is probably easier w SQL than it is through the interface.

Reply:
Now I definitely don't understand.  Do you want them to use sort of the same processing, but on a different data set?  I.e., maybe apply the same sequence of filters and grouping?  Do you have an example of what you're envisioning there (i.e., with actual table names and fake data)

Reply:
I don't think we can reuse views with different tables, at least not in any easy way that's worth doing for the alpha release.

I'll remove data sources being editable, it would be better to drop and create new columns.